ambry-api/src/main/java/com/github/ambry/quota/QuotaUtils.java

```diff
@@ -49,15 +52,20 @@ public static boolean isRequestResourceQuotaManaged(RestRequest restRequest) {
    *
    * @param restRequest {@link RestRequest} object.
    * @return QuotaResource extracted from headers of {@link RestRequest}.
-   * @throws RestServiceException if appropriate headers aren't found in the {@link RestRequest}.
+   * @throws QuotaException if appropriate headers aren't found in the {@link RestRequest}.
    */
-  public static QuotaResource getQuotaResourceId(RestRequest restRequest) throws RestServiceException {
-    Account account = RestUtils.getAccountFromArgs(restRequest.getArgs());
-    if (account.getQuotaResourceType() == QuotaResourceType.ACCOUNT) {
-      return QuotaResource.fromAccountId(account.getId());
-    } else {
-      Container container = RestUtils.getContainerFromArgs(restRequest.getArgs());
-      return QuotaResource.fromContainerId(account.getId(), container.getId());
+  public static QuotaResource getQuotaResource(RestRequest restRequest) throws QuotaException {
+    try {
+      Account account = RestUtils.getAccountFromArgs(restRequest.getArgs());
+      if (account.getQuotaResourceType() == QuotaResourceType.ACCOUNT) {
+        return QuotaResource.fromAccountId(account.getId());
+      } else {
+        Container container = RestUtils.getContainerFromArgs(restRequest.getArgs());
+        return QuotaResource.fromContainerId(account.getId(), container.getId());
+      }
+    } catch (RestServiceException rEx) {
+      LOGGER.error("Could not get quota resource for request.");
```

Comment:
can we add any request specific info in the log message, and the exception as well? The exception will be thrown away after this method.